instrumentation/google.golang.org/grpc/otelgrpc/config.go

```diff
@@ -193,3 +194,18 @@ func WithMetricAttributes(a ...attribute.KeyValue) Option {
 		}
 	})
 }
+
+type metricAttributesFnOption struct {
+	fn func(ctx context.Context) []attribute.KeyValue
+}
+
+func (o metricAttributesFnOption) apply(c *config) {
+	if o.fn != nil {
+		c.MetricAttributesFn = o.fn
+	}
+}
+
+// WithMetricAttributesFn returns an Option to add custom attributes to the metrics.
+func WithMetricAttributesFn(fn func(ctx context.Context) []attribute.KeyValue) Option {
+	return metricAttributesFnOption{fn: fn}
```

Comment:
We can directly use: `optionFunc` without the need to additionally define `metricAttributesFnOption`.